client/src/components/screens/PatientVariant/index.jsx

```diff
@@ -21,21 +22,24 @@ import { patientShape } from '../../../reducers/patient';
 import { variantShape } from '../../../reducers/variant';
 
 import Statement from '../../Query/Statement';
-import { fetchSchema, selectQuery, replaceQuery, replaceQueries, removeQuery, duplicateQuery, sortStatement, searchVariants, commitHistory, undo } from '../../../actions/variant';
+import { createNewQuery, fetchSchema, selectQuery, replaceQuery, replaceQueries, removeQuery, duplicateQuery, sortStatement, searchVariants, undo, addInstruction, removeInstruction, replaceInstruction, changeQueryTitle } from '../../../actions/variant';
 
 
 class PatientVariantScreen extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.handleCreateNewQuery = this.handleCreateNewQuery.bind(this);
     this.handleQuerySelection = this.handleQuerySelection.bind(this);
     this.handleQueryChange = this.handleQueryChange.bind(this);
     this.handleQueriesChange = this.handleQueriesChange.bind(this);
     this.handleQueriesRemoval = this.handleQueriesRemoval.bind(this);
     this.handleQueryDuplication = this.handleQueryDuplication.bind(this);
     this.handleStatementSort = this.handleStatementSort.bind(this);
-    this.handleCommitHistory = this.handleCommitHistory.bind(this);
     this.handleDraftHistoryUndo = this.handleDraftHistoryUndo.bind(this);
+    this.handleAddInstruction = this.handleAddInstruction.bind(this);
+    this.handleRemoveInstruction = this.handleRemoveInstruction.bind(this);
+    this.handleReplaceInstruction = this.handleReplaceInstruction.bind(this);
+    this.handleChangeQueryTitle = this.handleChangeQueryTitle.bind(this);
```

Comment:
Are these not all port of `handleQueryChange` ?

Reply:
Same thing here... No because, for example, you don't want to trigger an API call because you simply change the Title of the Query. Think of the redux sagas.